packages/react-hot-loader/src/utils.dev.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react'
+import hoistNonReactStatic from 'hoist-non-react-statics'
+import AppContainer from './AppContainer.dev'
+
+const getDisplayName = WrappedComponent =>
+  WrappedComponent.displayName || WrappedComponent.name || 'Component'
+
+const copyReactProp = (source, target) => {
+  hoistNonReactStatic(target, source)
+  target.displayName = `HotExported${getDisplayName(source)}`
+  target.WrappedComponent = source
+
+  target.propTypes = source.contextTypes
+  target.childContextTypes = source.childContextTypes
+  target.contextTypes = source.contextTypes
+}
+
+const makeHotExport = (sourceModule, getInstances) => {
+  const updateInstances = () => {
+    getInstances().forEach(inst => inst.forceUpdate())
+  }
+
+  const thenUpdateInstances = () => Promise.resolve(true).then(updateInstances)
+
+  if (sourceModule.hot) {
+    sourceModule.hot.accept(() => {
+      // Mark as self-accepted for Webpack
+      // Update instances for parsel
+      thenUpdateInstances()
+    })
+
+    // webpack way
+    if (sourceModule.hot.addStatusHandler) {
+      if (sourceModule.hot.status() === 'idle') {
+        sourceModule.hot.addStatusHandler(status => {
+          if (status === 'apply') {
+            thenUpdateInstances()
+          }
+        })
+      }
+    }
+  }
+}
+
+export const hotExported = (sourceModule, WrappedComponent, props = {}) => {
```

Comment:
The signature is not correct and not composable.

```js
hotExported(module)(WrappedComponent)
```

Also do we really need to specify "module"?

Reply:
The goal is to give RHL control over component's `module`. 
Next it will spinup HRM behavior to enable module self-acceptance, and next wrap Component with AppContainer.
So - this HOCs is doing 2 different things in real, but as long one shall have only one `hot` export per module - it is ok to merge all the stuff inside a single function.